llvm/test/CodeGen/RISCV/large-codemodel-sections.ll

```diff
@@ -0,0 +1,102 @@
+; RUN: llc -mtriple=riscv64 -mattr=+f,+zfh -target-abi=lp64f -code-model=large -verify-machineinstrs < %s \
+; RUN:   -filetype=obj -o - | llvm-objdump -htr - \
+; RUN:   | FileCheck %s -check-prefix=RV64I
+; RUN: llc -mtriple=riscv64 -mattr=+zfinx,+zhinx -target-abi=lp64 -code-model=large -verify-machineinstrs < %s \
+; RUN:   -filetype=obj -o - | llvm-objdump -htr - \
+; RUN:   | FileCheck %s -check-prefix=RV64I
+
+
+;; This tests that we are lowering large code model constants into `.text`
+;; constant pools, so that accessing them is close to `.text`, rather than
+;; far away in `.data`. The other choices are `.rodata` and `.data.rel.ro`,
+;; both of which may not be close enough to `.text` to be referenced.
+
+; RV64I-LABEL: Sections:
+; RV64I: .text 000000b4
```

Comment:
Are the section sizes the only thing showing where the constant pools are?

Reply:
I think the "best" thing showing where the constant pools are is the `-r` output, which says `RELOCATION RECORDS FOR [.text]:`, but that only works if there is actually a relocation - so true constants don't show up there.

Maybe an `objdump -dr` output would be better if we had taught objdump about mapping symbols for RISC-V.

Reply:
I checked with the newest pushed commit, and it definitely fails without my changes now.